### ISSUE TITLE
Reproduce results paper

### ISSUE THREAD

**Issue by BartvanMarrewijk** [ayaanzhaque/EC-GAN#3]
I have been trying to reproduce the results of the paper with 10% of the SVHN dataset. Unfortunately, the EC-GAN from the notebook does not reach the 91.15% as mentioned in the paper, it is even worse than the standard classifier. I might be doing something wrong, so I am curious to find out what. 

EC-GAN
Accuracy of the network on the 26032 test images: 85 ##
Train Accuracy: 98.4375

Standard classifier 
Accuracy of the network on the 26032 test images: 88 ##
Train Accuracy: 98.4375

Regarding the notebook/paper I also have following questions:
1. Which confidence threshold was used in the paper?
2. What was the reason to have an adversarial weight of 0.1. Was this value optimal? 
3. Why did you reload the dataset in the notebook twice? Since no seed is used, this would mean that the input data of both networks is different.
4.  What is the reason for a different LR settings for the classifier GAN and standard classifier?

**Comment by ayaanzhaque** [comment]
Hello,

Thanks for your interest in the work. I am quite surprised by your results. Let me answer your questions and we can work from there. 

1. 0.7
2. Yes, this value was found to be optimal in the ablative results section through tuning experiments.
3. I am not sure what you mean by this, can you refer to where in the notebook this is?
4. In the classifier GAN, we must make sure that the classifier is learning at a rate that is similar to the discriminator and generator, which is why the learning rate value is different.

**Comment by BartvanMarrewijk** [comment]
Thank for the quick response. 

1. Okay perfect, the default value in the notebook is 0.2 so, that might explain the different result in my runs. 
2. Thank you for the clarrification
3. In the data pre-processing segment step your are loading the data and apply a random permutation. However in the second cell of the Classifier segment, you are loading the data again. Also note that the datasizeConstant in this section is not the same  as in the pre-processing segment (0.25 vs 0.1)
```
#create subsets
dataSizeConstant = 0.25
valDataFraction = 0.1
subset = np.random.permutation([i for i in range(len(trainset))])
subTrain = subset[:int(len(trainset) * (dataSizeConstant))]
subTrainSet = datasets.SVHN("/content", split = "train", download = True, transform = transform_train)
subTrainLoader = DataLoader(subTrainSet, batch_size = batch_size, shuffle= False, num_workers= 2, sampler = torch.utils.data.SubsetRandomSampler(subTrain))

```
4. Thank you for the explanation. It makes sence, On the other way arround, both networks are trained for 100 epochs and both receive "high quality" data. 

The generated data by DCGAN contains some artifacts. What is your opinion about more powerfull GANS like StyleGAN?

**Comment by Lsjay3** [comment]
Hello, how did you set the confidence threshold ? You said the notebook defaults to 0.2, how did you got it ?
